HsSyck doesn't compile with gcc 14

GCC 14 [changed some warnings to errors](https://gcc.gnu.org/gcc-14/porting_to.html#warnings-as-errors), so HsSyck no longer builds due to `-Wimplicit-function-declaration`.

Full Log:

```
Running phase: setupCompilerEnvironmentPhase
Build with /nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6.
Running phase: unpackPhase
unpacking source archive /nix/store/4hmh4b4xlqx97y2xs3r67wnqqclaqlgy-HsSyck-0.53.tar.gz
source root is HsSyck-0.53
setting SOURCE_DATE_EPOCH to timestamp 1425014058 of file HsSyck-0.53/syck/yamlbyte.h
Running phase: patchPhase
Running phase: compileBuildDriverPhase
setupCompileFlags: -package-db=/build/tmp.lameFyDvQp/setup-package.conf.d -threaded
[1 of 2] Compiling Main             ( Setup.lhs, /build/tmp.lameFyDvQp/Main.o )
[2 of 2] Linking Setup
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
configureFlags: --verbose --prefix=/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53 --libdir=$prefix/lib/$compiler/lib --libsubdir=$abi/$libname --datadir=/nix/store/4x265mb1pr0gh0xnpa9zzwcab19l62j3-HsSyck-0.53-data/share/ghc-9.6.6 --docdir=/nix/store/h0zrfhwal2clkiqk7wj6myjnpz068zyb-HsSyck-0.53-doc/share/doc/HsSyck-0.53 --with-gcc=gcc --package-db=/build/tmp.lameFyDvQp/package.conf.d --ghc-option=-j2 --ghc-option=+RTS --ghc-option=-A64M --ghc-option=-RTS --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --enable-split-sections --enable-library-stripping --enable-executable-stripping --ghc-option=-haddock --extra-lib-dirs=/nix/store/wm1qn5jqrxpcjkc640gq8a90ns5gw3cn-ncurses-6.4.20221231/lib --extra-lib-dirs=/nix/store/r04wysx6yl7ys1rdv1b1lih6in1810r0-libffi-3.4.6/lib --extra-lib-dirs=/nix/store/wkysiv0sxqs6wn44nb46bbyig4plzsy4-elfutils-0.192/lib --extra-lib-dirs=/nix/store/ghb61mqc90qf1nx6kyahhhagicyxvcyg-gmp-with-cxx-6.3.0/lib
Using Parsec parser
Warning: HsSyck.cabal:31:30: version operators used. To use version operators
the package needs to specify at least 'cabal-version: >= 1.8'.
Configuring HsSyck-0.53...
Dependency base >=4 && <6: using base-4.18.2.1
Dependency bytestring >=0.9.0.1: using bytestring-0.11.5.3
Dependency hashtables: using hashtables-1.3.1
Dependency syb: using syb-0.7.2.4
Dependency utf8-string >=0.3: using utf8-string-1.0.2
Source component graph: component lib
Configured component graph:
    component HsSyck-0.53-7At0soVBbhO2B9mCeZ1JIp
        include base-4.18.2.1
        include bytestring-0.11.5.3
        include hashtables-1.3.1-Hrl6YQBaYQU1bKK0ZRVNLQ
        include syb-0.7.2.4-Ab7eJN2Bm6a4HiETVFK1pz
        include utf8-string-1.0.2-6PifKAdI4vLH9wwKyS8AsP
Linked component graph:
    unit HsSyck-0.53-7At0soVBbhO2B9mCeZ1JIp
        include base-4.18.2.1
        include bytestring-0.11.5.3
        include hashtables-1.3.1-Hrl6YQBaYQU1bKK0ZRVNLQ
        include syb-0.7.2.4-Ab7eJN2Bm6a4HiETVFK1pz
        include utf8-string-1.0.2-6PifKAdI4vLH9wwKyS8AsP
        Data.Yaml.Syck=HsSyck-0.53-7At0soVBbhO2B9mCeZ1JIp:Data.Yaml.Syck
Ready component graph:
    definite HsSyck-0.53-7At0soVBbhO2B9mCeZ1JIp
        depends base-4.18.2.1
        depends bytestring-0.11.5.3
        depends hashtables-1.3.1-Hrl6YQBaYQU1bKK0ZRVNLQ
        depends syb-0.7.2.4-Ab7eJN2Bm6a4HiETVFK1pz
        depends utf8-string-1.0.2-6PifKAdI4vLH9wwKyS8AsP
Using Cabal-3.10.3.0 compiled by ghc-9.6
Using compiler: ghc-9.6.6
Using install prefix: /nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53
Executables installed in:
/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53/bin
Libraries installed in:
/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53/lib/ghc-9.6.6/lib/x86_64-linux-ghc-9.6.6/HsSyck-0.53-7At0soVBbhO2B9mCeZ1JIp
Dynamic Libraries installed in:
/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53/lib/ghc-9.6.6/lib/x86_64-linux-ghc-9.6.6
Private executables installed in:
/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53/libexec/x86_64-linux-ghc-9.6.6/HsSyck-0.53
Data files installed in:
/nix/store/4x265mb1pr0gh0xnpa9zzwcab19l62j3-HsSyck-0.53-data/share/ghc-9.6.6/x86_64-linux-ghc-9.6.6/HsSyck-0.53
Documentation installed in:
/nix/store/h0zrfhwal2clkiqk7wj6myjnpz068zyb-HsSyck-0.53-doc/share/doc/HsSyck-0.53
Configuration files installed in:
/nix/store/zcrs4g6bk6sh3pl8yhn5f7k9phj0j03m-HsSyck-0.53/etc
No alex found
Using ar found on system at:
/nix/store/srcmmqi8kxjfygd0hyy42c8hv6cws83b-binutils-wrapper-2.43.1/bin/ar
No c2hs found
No cpphs found
No doctest found
Using gcc version 14.2.1 given by user at:
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/gcc
Using ghc version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/ghc
Using ghc-pkg version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/ghc-pkg-9.6.6
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.29.2 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/haddock-ghc-9.6.6
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/hpc-ghc-9.6.6
Using hsc2hs version 0.68.9 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/hsc2hs-ghc-9.6.6
Using hscolour version 1.25 found on system at:
/nix/store/v46cywpacz0q1snif3ac0crw31pq3h6z-hscolour-1.25/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/srcmmqi8kxjfygd0hyy42c8hv6cws83b-binutils-wrapper-2.43.1/bin/ld.gold
No pkg-config found
Using runghc version 9.6.6 found on system at:
/nix/store/pb9lljj3m0i6m3x30759k80pq5cf4sqi-ghc-9.6.6/bin/runghc-9.6.6
Using strip version 2.43 found on system at:
/nix/store/xcn9p4xxfbvlkpah7pwchpav4ab9d135-gcc-wrapper-14-20241116/bin/strip
Using tar found on system at:
/nix/store/9cwwj1c9csmc85l2cqzs3h9hbf1vwl6c-gnutar-1.35/bin/tar
No uhc found
Running phase: buildPhase
Preprocessing library for HsSyck-0.53..
Building library for HsSyck-0.53..
[1 of 1] Compiling Data.Yaml.Syck   ( dist/build/Data/Yaml/Syck.hs, dist/build/Data/Yaml/Syck.o, dist/build/Data/Yaml/Syck.dyn_o )
[1 of 1] Compiling Data.Yaml.Syck   ( dist/build/Data/Yaml/Syck.hs, dist/build/Data/Yaml/Syck.p_o )
gram.c: In function ‘syckparse’:

syck/gram.c:51:17: error:
     error: implicit declaration of function ‘sycklex’; did you mean ‘yylex’? [-Wimplicit-function-declaration]
       51 | #define yylex   sycklex
          |                 ^~~~~~~
   |
51 | #define yylex   sycklex
   |                 ^

gram.c:656:16: error:  note: in expansion of macro ‘yylex’

gram.c:1115:16: error:  note: in expansion of macro ‘YYLEX’
`gcc' failed in phase `C Compiler'. (Exit code: 1)
```

--- a/syck/gram.c
+++ b/syck/gram.c
@@ -141,7 +141,8 @@ typedef union YYSTYPE {
 # define YYSTYPE_IS_TRIVIAL 1
 #endif
 
-
+/* Forward declaration for sycklex */
+int sycklex( YYSTYPE *, void * );
 
 /* Copy the second part of user declarations.  */
 

